libminifi/src/Configure.cpp

```diff
@@ -71,12 +71,31 @@ const char *Configure::nifi_security_client_pass_phrase =
     "nifi.security.client.pass.phrase";
 const char *Configure::nifi_security_client_ca_certificate =
     "nifi.security.client.ca.certificate";
+const char *Configure::nifi_configuration_listener_pull_interval =
+    "nifi.configuration.listener.pull.interval";
+const char *Configure::nifi_configuration_listener_http_url =
+    "nifi.configuration.listener.http.url";
+const char *Configure::nifi_configuration_listener_rest_url =
+    "nifi.configuration.listener.rest.url";
+const char *Configure::nifi_configuration_listener_type =
+    "nifi.configuration.listener.type";
+const char *Configure::nifi_configuration_listener_need_ClientAuth =
+    "nifi.configuration.listener.need.ClientAuth";
+const char *Configure::nifi_configuration_listener_client_certificate =
```

Comment:
there are a lot of duplicate options here. it's possible that users will want to use different certs for different applications, which we should support; however, if consumers are using the same certificate perhaps we could make the configuration less onerous on them by tying configs together. this may make the eventual C2 impl easier to configure as well.